examples/immutable-values/immutable_values.bal

```diff
@@ -1,91 +1,50 @@
+    // Attempt to add an entry to the `map`, and trap the panic if it results in a panic.
     error? updateResult = trap addEntryToMap(m2, "intValTwo", 10);
     if (updateResult is error) {
         // An error should occur since `m2` is frozen.
         io:println("Error occurred on update: ",
-                   <string>updateResult.detail().message);
+                   <string>updateResult.detail()?.message);
     }
 
-    // Creates a `Department` record.
-    Department d = { name: "finance", id: 1100 };
+    // Now call `.cloneReadOnly()` on the immutable value `m2`.
+    map<string|int> m3 = m2.cloneReadOnly();
 
-    // Creates a `map` that may hold `anydata|error`-typed values.
-    // (`map<anydata|error>` belongs to `anydata`)
-    map<any> m3 = { stringVal: "str", intVal: 1, recVal: d };
-
-    // Attempts to freeze `m3`. The return type could now be an `error` since there is the possibility that a
-    // `map` constrained by the type `any` could have non-anydata values.
-    map<any>|error freezeResult = m3.freeze();
-    if (freezeResult is error) {
-        io:println("'.freeze()' failed for m3: ",
-                   <string>freezeResult.detail().message);
-    } else {
-        io:println("'.freeze()' successful for m3");
-    }
-
-    // Creates an `Employee` object.
-    Employee e = new("Anne");
-
-    // Now, creates a `map` that may hold `anydata|error` values, and add the non-anydata object `Employee` too.
-    map<any> m4 = { stringVal: "str", intVal: 1, objVal: e };
-
-    // Attempts to freeze `m4`.
-    freezeResult = m4.freeze();
-    if (freezeResult is error) {
-        io:println("'.freeze()' failed for m4: ",
-                   <string>freezeResult.detail().message);
-    } else {
-        io:println("'.freeze()' successful for m4");
-    }
+    // Reference equality checks for `m2` and `m3` should evaluate to true, due to no clones being created
+    // since `m2` is already an immutable value.
+    io:println("m2 === m3: ", m2 === m3);
 
     // An `is` check for a frozen value becomes an `is like` check.
     // In other words, storage type is not considered.
-    // Defines a `map` of the constraint type `string` or `int`, but with
+    // Define a `map` of the constraint type `string` or `int`, but with
     // values of the type `string` only.
     map<string|int> m5 = { valueType: "map", constraint: "string" };
-    // Freezes the `map`. The `.freeze()` attempt will be successful
-    // since the constraint is `anydata`. The frozen `map` only
-    // contains values of the type `string`.
-    var frozenVal = m5.freeze();
-    // Checking if the frozen value is of the type `map<string>` would
-    // evaluate to `true`.
+    // Make the map immutable. The resultant value would only
+    // contains values of the type `string`, and no values can now be
+    // added to the map.
+    var frozenVal = m5.cloneReadOnly();
+    // Checking if the frozen value is of the type `map<string>` thus
```

Comment:
```suggestion
    // Checkif the frozen value is of the type `map<string>`. Thus, it
```

Reply:
This change seems to introduce a disconnect here. This relates to the previous block.